modules/service/src/main/java/org/mule/runtime/module/service/internal/artifact/LibFolderClassLoaderConfigurationLoader.java

```diff
@@ -81,12 +84,27 @@ private List<URL> getServiceUrls(File rootFolder) {
     return urls;
   }
 
+  private List<URL> getServiceUrlsForJavaVersion(File rootFolder) {
+    List<URL> urls = new LinkedList<>();
+
+    final File libJavaVersionsDir = new File(rootFolder, LIB_DIR + "/java-versions");
+    if (libJavaVersionsDir.exists()) {
+      for (File file : libJavaVersionsDir.listFiles()) {
+        if (JVM_SPECIFICATION_VERSION >= parseInt(file.getName())) {
+          loadJarsFromFolder(urls, file);
+        }
+      }
```

Comment:
We should consider the issue that every version of a lib that is present for older JVMs than the one we are running with will be loaded. Suppose you are running with Java 17, and there are different versions of a lib for Java 11 and Java 17, both of them would be loaded.